src/controllers/reset/index.ts

```diff
@@ -0,0 +1,115 @@
+import { Get, Hidden, Produces, Route, SuccessResponse } from 'tsoa'
+import { inject, injectable, singleton } from 'tsyringe'
+
+import { Env } from '../../env.js'
+import { BadRequestError, InternalError } from '../../errors.js'
+import { Logger, type ILogger } from '../../logger.js'
+import Database from '../../models/db/index.js'
+import type { ConnectionRow } from '../../models/db/types.js'
+import VeritableCloudagent from '../../models/veritableCloudagent.js'
+
+@singleton()
+@injectable()
+// @Security('oauth2')
+@Route('/reset')
+@Hidden()
+@Produces('text/html')
+export class ResetController {
+  constructor(
+    private env: Env,
+    private db: Database,
+    private cloudagent: VeritableCloudagent,
+    @inject(Logger) private logger: ILogger
+  ) {
+    this.logger = logger.child({ controller: '/reset' })
+  }
+
+  // would be nice to restore to prior reset
+  public async isReset(): Promise<boolean> {
+    try {
+      const items: number[] = await Promise.all([
+        await this.db.get('connection', {}).then((res) => res.length),
+        await this.db.get('connection_invite', {}).then((res) => res.length),
+        await this.cloudagent.getCredentials().then((res) => res.length),
+        await this.cloudagent.getConnections().then((res) => res.length),
+      ])
+
+      this.logger.debug(
+        'items that have been not deleted:connection %s, inviute %s, credentials %s, connections %s',
+        items[0],
+        items[1],
+        items[2],
+        items[3]
+      )
+
+      return (
+        items.reduce((out: number, next: number) => {
+          return out + next
+        }, 0) === 0
+      )
+    } catch (err: unknown) {
+      this.logger.debug('%s', err)
+
+      return false
+    }
+  }
+
+  /**
+   * Retrieves the connection list page
+   */
+  @SuccessResponse(200)
+  @Get('/')
+  public async reset(): Promise<{ statusCode: number }> {
+    const DEMO_MODE = this.env.get('DEMO_MODE')
+    if (!DEMO_MODE) {
+      this.logger.debug('Instance is not in DEMO_MODE')
+
+      throw new BadRequestError('DEMO_MODE is false')
+    }
+
+    try {
+      const connections = (await this.db.get('connection')) || []
+      this.logger.debug('found connection %j', { connections })
+      const agentConnectionIds: string[] = connections
```

Comment:
I don't think we need to list these from the local db do we? Can't we just list `connections` and `credentials` from the cloudagent?

Reply:
agreed! and updated

Reply:
this hasn't been done

Reply:
should reflect in the latest 
```ts
      const connections: Connection[] = await this.cloudagent.getConnections()
      const credentials: Credential[] = await this.cloudagent.getCredentials()

```